packages/core/src/modules/connections/DidExchangeProtocol.ts

```diff
@@ -523,58 +523,56 @@ export class DidExchangeProtocol {
+        }
 
-      // JWS payload must be base64url encoded
-      const base64UrlPayload = base64ToBase64URL(didRotateAttachment.data.base64)
-      const signedDid = TypedArrayEncoder.fromBase64(base64UrlPayload).toString()
+        // JWS payload must be base64url encoded
+        const base64UrlPayload = base64ToBase64URL(didRotateAttachment.data.base64)
+        const signedDid = TypedArrayEncoder.fromBase64(base64UrlPayload).toString()
 
-      if (signedDid !== message.did) {
-        throw new CredoError(
-          `DID Rotate attachment's did ${message.did} does not correspond to message did ${message.did}`
-        )
-      }
+        if (signedDid !== message.did) {
+          throw new CredoError(
+            `DID Rotate attachment's did ${message.did} does not correspond to message did ${message.did}`
+          )
+        }
 
-      const { isValid, signerKeys } = await this.jwsService.verifyJws(agentContext, {
-        jws: {
-          ...jws,
-          payload: base64UrlPayload,
-        },
-        jwkResolver: ({ jws: { header } }) => {
-          if (typeof header.kid !== 'string' || !isDid(header.kid, 'key')) {
-            throw new CredoError('JWS header kid must be a did:key DID.')
-          }
+        const { isValid, signerKeys } = await this.jwsService.verifyJws(agentContext, {
+          jws: {
+            ...jws,
+            payload: base64UrlPayload,
+          },
+          jwkResolver: ({ jws: { header } }) => {
+            if (typeof header.kid !== 'string' || !isDid(header.kid, 'key')) {
+              throw new CredoError('JWS header kid must be a did:key DID.')
+            }
 
-          const didKey = DidKey.fromDid(header.kid)
-          return getJwkFromKey(didKey.key)
-        },
-      })
+            const didKey = DidKey.fromDid(header.kid)
+            return getJwkFromKey(didKey.key)
+          },
+        })
 
-      if (!isValid || !signerKeys.every((key) => invitationKeysBase58?.includes(key.publicKeyBase58))) {
-        throw new DidExchangeProblemReportError(
-          `DID Rotate signature is invalid. isValid: ${isValid} signerKeys: ${JSON.stringify(
-            signerKeys
-          )} invitationKeys:${JSON.stringify(invitationKeysBase58)}`,
-          {
-            problemCode: DidExchangeProblemReportReason.ResponseNotAccepted,
-          }
-        )
+        if (!isValid || !signerKeys.every((key) => invitationKeysBase58?.includes(key.publicKeyBase58))) {
+          throw new DidExchangeProblemReportError(
+            `DID Rotate signature is invalid. isValid: ${isValid} signerKeys: ${JSON.stringify(
+              signerKeys
+            )} invitationKeys:${JSON.stringify(invitationKeysBase58)}`,
+            {
+              problemCode: DidExchangeProblemReportReason.ResponseNotAccepted,
+            }
+          )
+        }
+      } else {
+        this.logger.warn(`Document does not contain didRotate`)
       }
```

Comment:
I don't think this change is correct. `extractResolvableDidDocument` is called when the did document is not `did:peer:1`. If we resolve the method, we REQUIRE the rotate attachment to be present. Could you explain a bit more why this change is needed?

Reply:
The test case I ran into was agents or mediators using did:peer:2 without the intention of doing a did rotate or `alsoKnownAs`. It also is a part of the didExchange RFC that the value is optional for any reason so for interoperability sake I think this should remain.

Reply:
If the did is not rotated from the invitation we can indeed ignore it. But if the did is rotated, not doing this check means the connection can be hijacked. I could send a response even though you created the invitation, because we don't require a signature from the invitation did. If the did is the same, then we don't have that issue and i'm ok with not requiring did rotate. But the current code doesn't take this into consideration

Reply:
I might be confused but the warn only happens if the attachment is not present, the else block is for the if statement on line 526 where it checks if the `didRotateAttachment` is defined. The signature is always checked if the rotate attachment is present. The else block with the warn was a suggestion made by @genaris when this PR was initially made.

Reply:
The check that is now in place always allows didRotate to not be present, meaning we could just remove the check. So we need to be more specific in checking when didRotate needs to be present: when you rotate your did from the invitation. Even though the Aries RFC mentions it's optional, it's super important for security to do this check. did_rotate was added specifically for this reason, and it's only optional to not introduce a breaking change in the RFC. Before did rotate was introduced, it was basically not possible to rotate a did from the invitation if you were not using diddoc~attach and sign it.

If the did is rotated from the invitation, either a signed diddoc~attach must be present OR a signed did_rotate~attach must be present